File: src/main/java/app/application/exception/ExceptionController.java
package app.application.exception;

import app.application.utils.DialogManager;
import org.springframework.stereotype.Service;

import javax.annotation.PostConstruct;

@Service
public class ExceptionController {

	private DialogManager dialogManager;

	public ExceptionController(DialogManager dialogManager) {
		this.dialogManager = dialogManager;
	}

	@PostConstruct
	public void error(){
		Thread.setDefaultUncaughtExceptionHandler((thread, throwable) -> dialogManager.openExceptionDialog(throwable));
	}

}

File: src/main/kotlin/app/application/exception/controller/ExceptionController.kt
package app.application.exception.controller

import app.application.spring.service.DialogManager
import org.springframework.stereotype.Service
import javax.annotation.PostConstruct

@Service
class ExceptionController(private val dialogManager: DialogManager) {

    @PostConstruct
    fun error() {
        Thread.setDefaultUncaughtExceptionHandler { _: Thread?, throwable: Throwable? -> dialogManager.openExceptionDialog(throwable as Exception?) }
    }

}